scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -4181,12 +4181,12 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
         else
           in.observeIndented()
 
-      if (nextIndented)
-        indentedOnOpen(templateStatSeq(enumCaseAllowed, secondaryConstructorAllowed))
-      else if (!enumCaseAllowed && isEndMarkerIntro(tokenPos))
+      if (!enumCaseAllowed && isEndMarkerIntro(tokenPos) && !nextIndented)
         (selfEmpty(), Nil)
-      else
-        syntaxError("expected template body", token)
+      else 
+        // https://docs.scala-lang.org/scala3/reference/other-new-features/indentation.html#optional-braces-around-template-bodies
+        // https://github.com/scalameta/scalameta/issues/3093
+        indentedOnOpen(templateStatSeq(enumCaseAllowed, secondaryConstructorAllowed))
```

Comment:
`indentedOnOpen` may be called only if it's definitely known that the current token is `Indentation.Indent`, and in this case it isn't.

I'd change to this:
```
      if (enumCaseAllowed)
        in.observeIndentedEnum()
      else
        in.observeIndented()

      if (acceptOpt[Indentation.Indent])
        indentedAfterOpen(templateStatSeq(enumCaseAllowed, secondaryConstructorAllowed))
```
and the rest could remain unchanged (`else if (!enumCaseAllowed...`).

Reply:
Thanks! that makes sense :+1: